commands/cli_nukeallresources_test.go

```diff
@@ -0,0 +1,547 @@
+package commands
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/gruntwork-io/cloud-nuke/aws"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/cloud-nuke/util"
+	"github.com/gruntwork-io/terratest/modules/collections"
+	"github.com/stretchr/testify/require"
+)
+
+// createIAMRoleArgs bundles up aws.CreateIAMRole args
+type createIAMRoleArgs struct {
+	awsSession               *session.Session
+	roleName                 string
+	roleDescription          string
+	assumeRolePolicyDocument string
+}
+
+// createIAMRolePolicyArgs bundles up aws.CreateIAMRolePolicy args
+type createIAMRolePolicyArgs struct {
+	awsSession     *session.Session
+	roleName       string
+	policyName     string
+	policyDocument string
+}
+
+// createTestResourcesArgs represents part of createTestResources args
+type createTestResourcesArgs struct {
+	resourceType string
+	name         string
+	tags         []map[string]string
+}
+
+func TestMain(m *testing.M) {
+	err := aws.SetEnvLogLevel()
+	if err != nil {
+		logging.Logger.Errorf("Invalid log level - %s", err)
+		os.Exit(1)
+	}
+	exitVal := m.Run()
+	os.Exit(exitVal)
+}
+
+func resourceExists(targetRegion string, targetResourceType string, targetResourceName string, account *aws.AwsAccountResources) bool {
+	for region, accountResources := range account.Resources {
+		if region != targetRegion {
+			continue
+		}
+		for _, resourceCollectionObj := range accountResources.Resources {
+			if resourceCollectionObj.ResourceName() == targetResourceType {
```

Comment:
Short answer: `ResourceName()` is a misnomer - it actually points to a resource's type. 


Long answer:

`accountResources` is of type 

```
type AwsAccountResources struct {
	Resources map[string]AwsRegionResource
}
```

where AwsRegionResource is 

```
type AwsRegionResource struct {
	Resources []AwsResources
}
```

and AwsResources is 

```
type AwsResources interface {
	ResourceName() string
	ResourceIdentifiers() []string
	MaxBatchSize() int
	Nuke(session *session.Session, identifiers []string) error
}
```

The `ResourceName` key in this type should ideally be called `ResourceType` because ResourceNames key in this is actually `ResourceIdentifiers`. All resources use `ResourceName` when they should mean `ResourceType` e.g. 

```

// ResourceName - the simple name of the aws resource
func (instance EC2Instances) ResourceName() string {
        return "ec2"
}
```

```

// ResourceName - the simple name of the aws resource
func (address EIPAddresses) ResourceName() string {
        return "eip"
}
```